modules/partition-replicator/src/integrationTest/java/org/apache/ignite/internal/partition/replicator/ItReplicaLifecycleTest.java

```diff
@@ -305,6 +321,94 @@ public void testEmptyReplicaListener() throws NodeStoppingException {
         assertEquals(-1, keyValueView.get(null, 1L));
     }
 
+    @Test
+    @Disabled("https://issues.apache.org/jira/browse/IGNITE-22410")
+    void testAlterReplicaTrigger() throws Exception {
+        Assignment replicaAssignment = (Assignment) AffinityUtils.calculateAssignmentForPartition(
+                nodes.stream().map(n -> n.name).collect(Collectors.toList()), 0, 1).toArray()[0];
+
+        Node node = getNode(replicaAssignment.consistentId());
+
+        placementDriver.setPrimary(node.clusterService.topologyService().localMember());
+
+        createZone(node, "test_zone", 2, 3);
```

Comment:
Let's assert after the every create/alter(Zone) that the result stable assignments of zone includes the current primary replica